transmart-rest-api/grails-app/controllers/org/transmartproject/rest/UserQuerySetController.groovy

```diff
@@ -0,0 +1,107 @@
+package org.transmartproject.rest
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.PathVariable
+import org.transmartproject.core.userquery.UserQuerySetDiff
+import org.transmartproject.core.userquery.UserQuerySetInstance
+import org.transmartproject.core.userquery.UserQuerySetResource
+import org.transmartproject.rest.misc.CurrentUser
+
+import static org.transmartproject.rest.misc.RequestUtils.checkForUnsupportedParams
+
+class UserQuerySetController {
+
+    @Autowired
+    CurrentUser currentUser
+
+    @Autowired
+    UserQuerySetResource userQuerySetResource
+
+    static responseFormats = ['json']
+
+    /**
+     * Scans for changes in results of the stored queries and updates stored sets:
+     * <code>/${apiVersion}/query_sets/scan</code>
+     *
+     * This endpoint should be called after loading, deleting or updating the data.
+     * Only available for administrators.
+     *
+     * @return number of sets that were updated, which is also a number of created querySets
+     */
+    def scan() {
+        Integer result = userQuerySetResource.scan(currentUser)
+        response.status = 201
+        respond([numberOfUpdatedSets: result])
+    }
+
+    /**
+     * Gets a list of query sets related to a specific query:
+     * <code>/${apiVersion}/query_sets/${queryId}</code>
+     *
+     * @param firstResult - first result - parameter required for pagination, by default equals 0
+     * @param numResults - number of results - parameter required for pagination
+     * @param queryId - id of the query
+     * @return list of querySets
+     */
+    def getSetsByQueryId(@PathVariable('queryId') Long queryId) {
+        checkForUnsupportedParams(params, ['queryId', 'firstResult', 'numResults'])
+        int firstResult = params.firstResult ?: 0
+        def numResults = params.numResults as Integer
+
+        List<UserQuerySetInstance> querySetInstances = userQuerySetResource.getSetInstancesByQueryId(queryId, currentUser,
```

Comment:
Why not just fetch a user query set, which has a set of patients/patient ids or object ids? Something similar happens but then with lists of set entries that are grouped, but I think the contract of the service call should be different.

Reply:
When I tried to fetch the user query set directly, querySetInstances were empty (same for querySetDiffs) and I couldn't make it working, that is why I added a different solution with grouping.

Reply:
Well, but this is really hard to read. And what is the relation between `getSetsByQueryId` and  `getDiffsByQueryId`. What are their types? Etc.

Reply:
These wrappers are really not needed. The data showcase is a better example of how to write a controller than most of Transmart, so please don't be inspired by the other controllers here.